src/poliastro/core/iod.py

```diff
@@ -293,6 +332,99 @@ def _compute_y(x, ll):
     return np.sqrt(1 - ll ** 2 * (1 - x ** 2))
 
 
+@jit
+def _check_bounds(psi, psi_new, psi_up, psi_low, tof_new, tof):
+    """Updates psi_new, psi_up, psi_low
+
+    "Check if newton guess for psi is outside bounds (too steep a slope)
+     and update values accordingly"
+
+    """
+    if np.abs(psi_new) > psi_up or psi_new < psi_low:
+        # lower_bound
+        steep_condition_1 = tof_new < tof and psi > psi_low
+        psi_low = psi_low + (psi - psi_low) * steep_condition_1
+
+        # upper_bound
+        steep_condition_2 = tof_new > tof and psi < psi_up
+        psi_up = psi_up + (psi - psi_up) * steep_condition_2
+
+        psi_new = (psi_up + psi_low) / 2
+
+    return psi_new, psi_low, psi_up
```

Comment:
The only difference previous code and this one is, the return statement. This time we return regardless of ```if(condition)```.